src/auth/ha-authorize.ts

```diff
@@ -145,7 +133,7 @@ class HaAuthorize extends litLocalizeLiteMixin(LitElement) {
         response.status === 400 &&
         authProviders.code === "onboarding_required"
       ) {
-        location.href = "/?";
+        location.href = `/onboarding.html${location.search}`;
```

Comment:
This is where all the magic is happening. It passes all parameters passed to the authorize page to the onboarding so they can access it at the end.